### PR TITLE
jwt auth middleware

### PR DESCRIPTION
**Pull request by pandres95**

### CODE REVIEW

**Comment by pandres95** [comment]
marked as draft since it's giving out issues when running on docker

File: merchant/src/utils/jwt.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+pub fn unauthorized_error() -> Response {
+    let mut res = Response::new(401);
+    res.set_body("Unauthorized");
+
+    res
+}
+
+pub fn jwt_middleware<'a, State: Clone + Send + Sync + 'static>(
```
Comment (olanod) [inline merchant/src/utils/jwt.rs:31]: Can't the function be `async` to not have to use the `Pin<Box<_>>`?

File: merchant/src/catalog/models.rs [hunk 2]
```diff
@@ -68,18 +68,32 @@ pub enum Time {
...
+            url: "".to_owned()
+        }
+    }
+}
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
 pub struct Item {
     pub category: ItemCategory,
     pub tags: Vec<String>,
     pub name: String,
+    #[serde(default = "default_images")]
```
Comment (olanod) [inline merchant/src/catalog/models.rs:84]: Do we need this if the default is an empty vec?

File: merchant/src/catalog/backend.rs [hunk 3]
```diff
@@ -642,12 +646,18 @@ impl CatalogService for CatalogSQLService {
             .pool
             .acquire()
             .await
-            .map_err(|_| CatalogError::DatabaseError)?;
+            .map_err(|err| {
+                println!("{:?}", err);
```
Comment (olanod) [inline merchant/src/catalog/backend.rs:650]: ✂️ ... in a separate PR we could introduce the `log` crate replacing all occurences of `println!`

File: merchant/src/catalog/backend.rs [hunk 4]
```diff
@@ -642,12 +646,18 @@ impl CatalogService for CatalogSQLService {
...
+                CatalogError::DatabaseError
+            })?;
 
         let result: Vec<CatalogObjectRow> = bind_query_as(sqlx::query_as(&sql), &values)
             .fetch_all(&mut pool)
             .await
-            .map_err(|_| CatalogError::DatabaseError)?;
+            .map_err(|err| {
+                println!("{:?}", err);
```
Comment (olanod) [inline merchant/src/catalog/backend.rs:658]: ✂️

**Pull request closed without merging**